datastore_mongo.go

```diff
@@ -85,6 +85,23 @@ func (db *DataStoreMongo) GetDevice(id DeviceID) (*Device, error) {
 	return &res, nil
 }
 
+func (db *DataStoreMongo) DeleteDevice(id DeviceID) error {
+	s := db.session.Copy()
+	defer s.Close()
+
+	filter := bson.M{"id": id}
```

Comment:
not sure, but we use upsert, and there's this bit of warning in mongo docs:

https://docs.mongodb.com/manual/reference/method/db.collection.update/#use-unique-indexes

so I'm leaning towards 'yes we do'. thoughts? plus - where to put it - just stick it into code, or actually create a migration?